guardian/core.py

```diff
@@ -123,3 +144,78 @@ def get_local_cache_key(self, obj):
         ctype = ContentType.objects.get_for_model(obj)
         return (ctype.id, obj.pk)
 
+    def prefetch_perms(self, objects):
+        """
+        Prefetches the permissions for objects in ``objects`` and puts them in the cache.
+
+        :param objects: Iterable of Django model objects
+
+        """
+        if self.user and not self.user.is_active:
+            return []
+
+        User = get_user_model()
+        pks, model, ctype = _get_pks_model_and_ctype(objects)
+
+        if self.user and self.user.is_superuser:
+            perms = list(chain(
+                *Permission.objects
+                .filter(content_type=ctype)
+                .values_list("codename")))
+
+            for pk in pks:
+                key = (ctype.id, pk)
+                self._obj_perms_cache[key] = perms
+
+            return True
+
+        group_model = get_group_obj_perms_model(model)
+
+        group_filters = {
+            'object_pk__in': pks
+        }
+
+        if self.user:
+            fieldname = 'group__%s' % (
+                User.groups.field.related_query_name(),
+            )
+            group_filters.update({fieldname: self.user})
+        else:
+            group_filters = {'group': self.group}
+
+        if group_model.objects.is_generic():
+            group_filters.update({
+                'content_type': ctype,
+            })
+
+        if self.user:
+            model = get_user_obj_perms_model(model)
+            user_filters = {
+                'user': self.user,
+                'object_pk__in': pks
+            }
+
+            if model.objects.is_generic():
+                user_filters.update({
+                    'content_type': ctype,
+                })
+
+            perms_qs = Permission.objects.filter(content_type=ctype)
+            # Query user and group permissions separately and then combine
+            # the results to avoid a slow query
+            user_perms_qs = model.objects.filter(**user_filters).select_related('permission')
+            group_perms_qs = group_model.objects.filter(**group_filters).select_related('permission')
+            perms = chain(user_perms_qs, group_perms_qs)
+        else:
+            perms = chain(
+                *perms_qs.filter(**group_filters).select_related('permission')
+            )
+
+        for perm in perms:
+            key = (ctype.id, int(perm.object_pk))
```

Comment:
I would like mark that in line 167 key you generate by ``` (ctype.id, pk)```. I'm not sure this is correct to cast ```object_pk``` to int. CharField can have set ```primary_key=True``, so it will be ```object_pk```. In this case it will raises exceptions.